java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -834,4 +836,100 @@ public void expireAt_pexpireAt_and_ttl_with_non_existing_key(BaseClient client)
         assertEquals(-2L, client.ttl(key).get());
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zadd_and_zaddIncr(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        Map<String, Double> membersScores = Map.of("one", 1.0, "two", 2.0, "three", 3.0);
+
+        assertEquals(3, client.zadd(key, membersScores).get());
+        assertEquals(3.0, client.zaddIncr(key, "one", 2.0).get());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zadd_and_zaddIncr_with_NX_XX(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        Map<String, Double> membersScores = Map.of("one", 1.0, "two", 2.0, "three", 3.0);
+
+        ZaddOptions onlyIfExistsOptions =
+                ZaddOptions.builder()
+                        .conditionalChange(ZaddOptions.ConditionalChange.ONLY_IF_EXISTS)
+                        .build();
+        ZaddOptions onlyIfDoesNotExistOptions =
+                ZaddOptions.builder()
+                        .conditionalChange(ZaddOptions.ConditionalChange.ONLY_IF_DOES_NOT_EXIST)
+                        .build();
+
+        assertEquals(0, client.zadd(key, membersScores, onlyIfExistsOptions).get());
+        assertEquals(3, client.zadd(key, membersScores, onlyIfDoesNotExistOptions).get());
+        assertEquals(null, client.zaddIncr(key, "one", 5, onlyIfDoesNotExistOptions).get());
+        assertEquals(6, client.zaddIncr(key, "one", 5, onlyIfExistsOptions).get());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zadd_and_zaddIncr_with_GT_LT(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        Map<String, Double> membersScores = new LinkedHashMap<>();
+        membersScores.put("one", -3.0);
+        membersScores.put("two", 2.0);
+        membersScores.put("three", 3.0);
+
+        assertEquals(3, client.zadd(key, membersScores).get());
+        membersScores.put("one", 10.0);
+
+        ZaddOptions scoreGreaterThanOptions =
+                ZaddOptions.builder()
+                        .updateOptions(ZaddOptions.UpdateOptions.SCORE_GREATER_THAN_CURRENT)
+                        .build();
+        ZaddOptions scoreLessThanOptions =
+                ZaddOptions.builder()
+                        .updateOptions(ZaddOptions.UpdateOptions.SCORE_LESS_THAN_CURRENT)
+                        .build();
+
+        assertEquals(1, client.zadd(key, membersScores, scoreGreaterThanOptions, true).get());
+        assertEquals(0, client.zadd(key, membersScores, scoreLessThanOptions, true).get());
+        assertEquals(7, client.zaddIncr(key, "one", -3, scoreLessThanOptions).get());
+        assertEquals(null, client.zaddIncr(key, "one", -3, scoreGreaterThanOptions).get());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zadd_illegal_arguments(BaseClient client) {
```

Comment:
Please add tests where key does not exist and where key exists, but not a set

Reply:
When a key does not exist and error is not thrown. Added the wrong type test.